src/app/pages/CourseDetail/CourseDetail.tsx

```diff
@@ -29,7 +45,11 @@ function CourseDetail(): JSX.Element {
         </span>
         {priority && <span>{priorities[priority]?.icon}</span>}
         <span className={styles.courseName}>
-          <CourseDetailHeader />
+          <CourseDetailHeader
+            imgSrc="images/basketball.png"
+            name={filteredCourse[0].courseName}
```

Comment:
On first render, `filteredCourse[0]` is `undefined` (because the initial value of your `courses` array is `[]`).
This leads to an error like  `courseName of undefined`.

You should check if the course exists or is still loading. You could make use of [conditional rendering](https://reactwithhooks.netlify.app/docs/conditional-rendering.html).